### ISSUE TITLE
Querystring schema shorthand breaks usage of 

### ISSUE THREAD

**Issue by SkeLLLa** [fastify/fastify#1042]
Querysting schema shorthand applies always when `properties` property isn't set directly.

#### What are you trying to achieve or the steps to reproduce?

```js
fastify.addSchema({
      $id: 'mySchemaFilter',
      type: 'object',
      required: ['limit', 'skip'],
      allOf: [
        {
          type: 'object',
          properties: {
            limit: {type: 'integer', default: 30},
            skip: {type: 'integer', default: 0},
          },
        },
        'myParentSchema#',
      ],
    })
 fastify.get(
    '/',
    {
      schema: {
        description: `Some get`,
        summary: 'Summary',
        querystring: 'mySchemaFilter#',
        response: {
          [HTTP_CODE.OK]: {
            type: 'array',
            items: 'myParentSchema#',
          },
          [HTTP_CODE.NOT_FOUND]: 'errorNotFound#',
          [HTTP_CODE.UNAUTHORIZED]: 'errorUnauthorized#',
          default: 'error#',
        },
      },
    }, () => {}
  )
}
```

#### What was the result you received?

```
Error: schema is invalid: data.properties['$id'] should be object,boolean
```

#### What did you expect?

Schema `'mySchemaFilter#'` successfully applied to querystring

#### Context

* *node version*: 9
* *fastify version*: 1.8.0

**Comment by mcollina** [comment]
Is allOf supported by ajv itself? Have you tried specifying a single schema
defined in the route (non inheritance)?

Can you please include a full example and sample requests (ideally using
fastify.inject) to test?
Il giorno dom 15 lug 2018 alle 15:52 Alexander Kureniov <
notifications@github.com> ha scritto:

**Comment by SkeLLLa** [comment]
@mcollina Ajv supports `allOf`. However actually `allOf` support in fastify is done in https://github.com/fastify/fast-json-stringify/blob/4aaaff670f4437cdf8070370ad1363105b4490e0/index.js#L469.

I don't think that example requests will help, because an application will not start with such schemas and will crash with error described above.

So the main problem I think is if `properties` prop missing in `querystring` object it starts treating it like a shorthand.

Here's isolated issue:
```js
const fastify = require('fastify');
const app = fastify({
  logger: {
    level: 'info',
    prettyPrint: true,
  },
  ignoreTrailingSlash: true,
});
app.addSchema({
  $id: 'first',
  type: 'object',
  properties: {
    first: {
      type: 'number',
    },
  },
});

app.addSchema({
  $id: 'second',
  type: 'object',
  allOf: [
    {
      type: 'object',
      properties: {
        second: {
          type: 'number',
        },
      },
    },
    'first#',
  ],
});

app.addSchema({
  $id: 'third',
  type: 'object',
  allOf: [
    {
      type: 'object',
      properties: {
        third: {
          type: 'number',
        },
      },
    },
    'second#',
  ],
});

app.get(
  '/',
  {
    schema: {
      description: `get third`,
      querystring: 'third#',
      response: {
        200: 'third#',
      },
    },
  },
  async function(request, reply) {
    return {ok: 'ok'};
  }
);


(async () => {
  try {
    await Promise.all([
      app.listen(3333),
      app.ready(),
    ]);
  } catch (ex) {
    console.log(ex);
    process.exit(1);
  }
})();

```

**Comment by mcollina** [comment]
Can you please include a full stacktrace then? It was not clear from the
original description.
Il giorno dom 15 lug 2018 alle 17:08 Alexander Kureniov <
notifications@github.com> ha scritto:

**Comment by SkeLLLa** [comment]
```
Error: schema is invalid: data.properties['$id'] should be object,boolean
    at Ajv.validateSchema (/app/node_modules/ajv/lib/ajv.js:183:16)
    at Ajv._addSchema (/app/node_modules/ajv/lib/ajv.js:312:10)
    at Ajv.compile (/app/node_modules/ajv/lib/ajv.js:112:24)
    at build (/app/node_modules/fastify/lib/validation.js:62:34)
    at afterRouteAdded (/app/node_modules/fastify/fastify.js:565:9)
    at Object._encapsulateThreeParam (/app/node_modules/avvio/boot.js:360:7)
    at Boot.callWithCbOrNextTick (/app/node_modules/avvio/boot.js:308:5)
    at Boot._after (/app/node_modules/avvio/boot.js:212:26)
    at Plugin.exec (/app/node_modules/avvio/plugin.js:45:17)
    at Boot.loadPlugin (/app/node_modules/avvio/plugin.js:125:10)

```

**Comment by mcollina** [comment]
Would you like to send a PR with the fix?

**Comment by delvedor** [comment]
Closed in #1050
